src/lib/builtins.pl

```diff
@@ -26,42 +26,120 @@
                      write_canonical/1, write_canonical/2,
                      write_term/2, write_term/3, writeq/1, writeq/2]).
 
+/** Builtin predicates
+
+This library, unlike the rest, is loaded by default and it exposes the most fundamental and general
+predicates of the Prolog system under the ISO standard. Basic operators, metaprogramming, exceptions,
+internal settings and basic I/O are all here.
+*/
+
 
 % unify.
+
+%% =(?X, ?Y)
+%
+% Unify two variables. This is the most basic operation of Prolog.
+% Unification also happens when doing head matching in a rule.
 X = X.
 
+%% true.
+%
+% Always succeeds
 true.
 
+%% false.
+%
+% Always fails
 false :- '$fail'.
 
 
 % These are stub versions of call/{1-9} defined for bootstrapping.
 % Once Scryer is bootstrapped, each is replaced with a version that
 % uses expand_goal to pass the expanded goal along to '$call'.
 
+
+%% call(Goal).
+%
+% Execute the Goal. Typically used when the Goal is not known at compile time.
 call(_).
 
+%% call(Goal, ExtraArg1).
+%
+% Execute the Goal with ExtraArg1 appended to the argument list. For example:
+%
+%     ?- call(format("~s~n"), ["Alain Colmerauer"]).
+%     Alain Colmerauer
+%        true.
+%
+% Which is equivalent to: `format("~s~n", ["Alain Colmerauer"]).`
 call(_, _).
 
+%% call(Goal, ExtraArg1, ExtraArg2).
+%
+% Execute Goal with ExtraArg1 and ExtraArg2 appended to the argument list.
 call(_, _, _).
 
+%% call(Goal, ExtraArg1, ExtraArg2, ExtraArg3).
+%
+% Execute Goal with ExtraArg1, ExtraArg2 and ExtraArg3 appended to the argument list.
 call(_, _, _, _).
 
+%% call(Goal, ExtraArg1, ExtraArg2, ExtraArg3, ExtraArg4).
+%
+% Execute Goal with ExtraArg1, ExtraArg2, ExtraArg3 and ExtraArg4 appended to the argument list.
 call(_, _, _, _, _).
 
+%% call(Goal, ExtraArg1, ExtraArg2, ExtraArg3, ExtraArg4, ExtraArg5).
+%
+% Execute Goal with ExtraArg1, ExtraArg2, ExtraArg3, ExtraArg4 and ExtraArg5 appended to the argument list.
 call(_, _, _, _, _, _).
 
+%% call(Goal, ExtraArg1, ExtraArg2, ExtraArg3, ExtraArg4, ExtraArg5, ExtraArg6).
+%
+% Execute Goal with ExtraArg1, ExtraArg2, ExtraArg3, ExtraArg4, ExtraArg5 and ExtraArg6 appended
+% to the argument list.
 call(_, _, _, _, _, _, _).
 
+%% call(Goal, ExtraArg1, ExtraArg2, ExtraArg3, ExtraArg4, ExtraArg5, ExtraArg6, ExtraArg7).
+%
+% Execute Goal with ExtraArg1, ExtraArg2, ExtraArg3, ExtraArg4, ExtraArg5, ExtraArg6 and ExtraArg7
+% appended to the argument list.
 call(_, _, _, _, _, _, _, _).
 
+%% call(Goal, ExtraArg1, ExtraArg2, ExtraArg3, ExtraArg4, ExtraArg5, ExtraArg6, ExtraArg7, ExtraArg8).
+%
+% Execute Goal with ExtraArg1, ExtraArg2, ExtraArg3, ExtraArg4, ExtraArg5, ExtraArg6, ExtraArg7 and
+% ExtraArg8, appended to the argument list.
 call(_, _, _, _, _, _, _, _, _).
 
 
 :- meta_predicate catch(0, ?, 0).
 
 % flags.
 
+%% current_prolog_flag(Flag, Value)
+%
+% Returns the current Value of several flags in the running system. A flag is a setting which value affects
```

Comment:
What is "Returns"? 

I suggest the phrasing from the ISO standard to start this paragraph:

<blockquote>
current_prolog_flag(Flag, Value) is true iff Flag
is a flag supported by the processor, and Value is the
value currently associated with it.
</blockquote>